pkgs/development/tools/build-managers/dub/default.nix

```diff
@@ -1,4 +1,6 @@
-{ lib, stdenv, fetchFromGitHub, curl, dmd, libevent, rsync }:
+{ lib, stdenv, fetchFromGitHub, curl, libevent, rsync, ldc ? null, dcompiler ? ldc }:
```

Comment:
How many dcompiler are there? If there are only two we should probably use a boolean flag or two. Also we should probably check against a list of known to work compilers with asserts instead of bash code. This tells us immediately if the build fails.

Reply:
There are 3 D compilers (sharing the dmd frontend):

- dmd (the reference upstream compiler)
- gdc (GCC-based) and 
- ldc (LLVM-based)
https://dlang.org/download.html

Each of the 3 have their strong points:

- D generally but dmd in particular has the reputation of being extremely fast at compiling (so much so that some use it for scripting)
- gdc and ldc produce faster (and usually smaller) binaries
- gdc and ldc target more architectures
- until recently gdc has lagged in having up to date versions of D (while ongoing integration work takes place), but it appears this may be about to change as it is reported today via Hacker News that D is now an official part of GCC
  - https://news.ycombinator.com/item?id=26332312
  - https://gcc.gnu.org/git/?p=gcc.git;a=tree;f=gcc/d;h=557fd0c0954abe5d45c8c394b7e6416710c72e59;hb=refs/heads/master
- ldc is pretty much in sync with dmd being updated within days